src/ring_theory/ideal/operations.lean

```diff
@@ -1139,29 +1140,41 @@ end
 
 section quotient_algebra
 
-/-- The `R`-algebra structure on `R/I` -/
-instance {I : ideal R} : algebra R (ideal.quotient I) := (ideal.quotient.mk I).to_algebra
+variables (R) {A : Type*} [comm_ring A] [algebra R A]
 
-/-- The canonical morphism `R →ₐ[R] I.quotient`, for `I` an ideal of `R`, as morphism of
-`R`-algebras. -/
-def quotient.mkₐ (I : ideal R) : R →ₐ[R] I.quotient :=
+/-- The `R`-algebra structure on `A/I` for an `R`-algebra `A` -/
+instance {I : ideal A} : algebra R (ideal.quotient I) :=
+(ring_hom.comp (ideal.quotient.mk I) (algebra_map R A)).to_algebra
+
+/-- The canonical morphism `A →ₐ[R] I.quotient` as morphism of `R`-algebras, for `I` an ideal of
+`A`, where `A` is an `R`-algebra. -/
+def quotient.mkₐ (I : ideal A) : A →ₐ[R] I.quotient :=
 ⟨λ a, submodule.quotient.mk a, rfl, λ _ _, rfl, rfl, λ _ _, rfl, λ _, rfl⟩
 
-lemma quotient.mkₐ_to_ring_hom (I : ideal R) :
-  (quotient.mkₐ I).to_ring_hom = ideal.quotient.mk I := rfl
+lemma quotient.alg_map (I : ideal A) :
+  algebra_map R I.quotient = (algebra_map A I.quotient).comp (algebra_map R A) :=
+by simp only [ring_hom.algebra_map_to_algebra, ring_hom.comp_id]
+
+instance {I : ideal A} : is_scalar_tower R A (ideal.quotient I) :=
+is_scalar_tower.of_algebra_map_eq' (quotient.alg_map R I)
```

Comment:
```suggestion
lemma quotient.alg_map_eq (I : ideal A) :
  algebra_map R I.quotient = (algebra_map A I.quotient).comp (algebra_map R A) :=
by simp only [ring_hom.algebra_map_to_algebra, ring_hom.comp_id]

instance {I : ideal A} : is_scalar_tower R A (ideal.quotient I) :=
is_scalar_tower.of_algebra_map_eq' (quotient.alg_map_eq R I)
```
since otherwise `alg_map` seems too much like an alias for `algebra_map` rather than a lemma about it